scst/src/dev_handlers/scst_vdisk.c

```diff
@@ -6432,6 +6432,100 @@ static void vdisk_task_mgmt_fn_done(struct scst_mgmt_cmd *mcmd,
 	return;
 }
 
+static void vdisk_bio_ext_copy_remap(struct scst_cmd *cmd,
+	struct scst_ext_copy_seg_descr *seg)
+{
+	struct block_device *bdev_in, *bdev_out;
+	loff_t pos_in, pos_out, pos_shift = 0;
+	size_t len, chunk, rem, ret, d_count = 0;
+	uint64_t size_in, size_out, blocksize;
+	struct scst_ext_copy_data_descr *d;
+	struct scst_vdisk_dev *src_virt_dev;
+	struct scst_vdisk_dev *dst_virt_dev;
+	uint64_t in_shift;
+	uint64_t out_shift;
+
+	if (!seg || !seg->src_tgt_dev || !seg->dst_tgt_dev)
+		return;
+
+	src_virt_dev = seg->src_tgt_dev->dev->dh_priv;
+	dst_virt_dev = seg->dst_tgt_dev->dev->dh_priv;
+
+	if (!src_virt_dev->blockio || !dst_virt_dev->blockio)
+		goto out;
+
+	if (strcmp(seg->dst_tgt_dev->dev->handler->name, "vdisk_blockio"))
+	    goto out;
```

Comment:
I am not sure we can dereference `dst_virt_dev` before checking its type.  I think the `if (strcmp(...` should be done earlier.  Also there seems to be something wrong with indentation.

Reply:
Thanks, updated!